### PR TITLE
Added fixes for issue #971(`Mounted volume can be deleted`)

### PR DESCRIPTION
**Pull request by click2cloud-lamda**
Fixed issue #971 

In this PR, if the volume is mounted, it doesn't allow the volume to be removed.

```
D:\GolandProjects\github.com\nerdctl\cmd\nerdctl>nerdctl run -d -v testVolume:C:\src mcr.microsoft.com/windows/nanoserver:20H2
23043c648edcfd53309d2d365676072bb5b6e24cc06e4a5c61769860a1015edf

D:\GolandProjects\github.com\nerdctl\cmd\nerdctl>nerdctl volume rm testVolume
time="2022-04-11T16:45:33+05:30" level=error msg="Remove Volume: \"testVolume\" failed" error="Volume \"testVolume\" is in use"

```

### CODE REVIEW

File: cmd/nerdctl/volume_rm.go [hunk 1]
```diff
@@ -39,14 +43,51 @@ func newVolumeRmCommand() *cobra.Command {
...
+			}
+			err = json.Unmarshal(n.Container.Spec.Value, &mount)
+			if err != nil {
+				return err
+			}
+			volGet, _ := volStore.Get(name)
+			if found = checkVolume(&mount.Mounts, volGet.Mountpoint); found {
+				found = true
+				var verr error
+				verr = fmt.Errorf("%s %q %s", "Volume", name, "is in use")
```
Comment (AkihiroSuda) [inline cmd/nerdctl/volume_rm.go:75]: No need to use %s

File: cmd/nerdctl/volume_rm.go [hunk 2]
```diff
@@ -39,14 +43,51 @@ func newVolumeRmCommand() *cobra.Command {
 }
 
 func volumeRmAction(cmd *cobra.Command, args []string) error {
+	client, ctx, cancel, err := newClient(cmd)
+	if err != nil {
+		return err
+	}
+	defer cancel()
+	containers, _ := client.Containers(ctx)
```
Comment (AkihiroSuda) [inline cmd/nerdctl/volume_rm.go:51]: catch err

File: cmd/nerdctl/volume_rm.go (hunk 1 above)
Comment (click2cloud-lamda) [reply cmd/nerdctl/volume_rm.go]: Okay

File: cmd/nerdctl/volume_rm.go [hunk 3]
```diff
@@ -39,14 +43,52 @@ func newVolumeRmCommand() *cobra.Command {
...
+		}
+		if !found {
+			// volumes that are not mounted to any container
+			removednames = append(removednames, name)
+		}
+	}
+	if removednames != nil {
+		rNames, err := volStore.Remove(removednames)
+		if err != nil {
+			panic(err)
```
Comment (Zheaoli) [inline cmd/nerdctl/volume_rm.go:89]: I think just return error directly would be better

File: cmd/nerdctl/volume_rm.go [hunk 4]
```diff
@@ -39,14 +43,52 @@ func newVolumeRmCommand() *cobra.Command {
...
+	if err != nil {
+		return err
+	}
+	defer cancel()
+	containers, err := client.Containers(ctx)
+	if err != nil {
+		return err
+	}
+	var mount *specs.Spec
+	var removednames []string
```
Comment (Zheaoli) [inline cmd/nerdctl/volume_rm.go:56]: `var removedNames` would be better?

Comment (click2cloud-lamda) [reply cmd/nerdctl/volume_rm.go:56]: I used it for volumes that have to be deleted.

File: cmd/nerdctl/volume_rm.go (hunk 3 above)
Comment (click2cloud-lamda) [reply cmd/nerdctl/volume_rm.go:89]: Ill change it.

**Review by AkihiroSuda — APPROVED** [review]

**Pull request merged**